From sony.chacko@qlogic.com  Thu Jun 23 22:59:43 2011
From: Sony Chacko <sony.chacko@qlogic.com>
Date: Thu, 10 Mar 2011 23:50:02 +0000
Subject: [PATCH 48/56] netxen: Notify firmware of Flex-10 interface down
Git-commit: d478af0d6bb38c971d1172e99c5046025a9167c9
Patch-mainline: v2.6.39-rc1
References: bnc#698269, FATE#311464

Notify firmware when a Flex-10 interface is brought down
so that virtual connect manager can display the correct link status.

Signed-off-by: Sony Chacko <sony.chacko@qlogic.com>
Signed-off-by: Amit Kumar Salecha <amit.salecha@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/netxen/netxen_nic_main.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/net/netxen/netxen_nic_main.c b/drivers/net/netxen/netxen_nic_main.c
index e0a315e..8261660 100644
--- a/drivers/net/netxen/netxen_nic_main.c
+++ b/drivers/net/netxen/netxen_nic_main.c
@@ -1031,6 +1031,9 @@ __netxen_nic_down(struct netxen_adapter *adapter, struct net_device *netdev)
 	netif_carrier_off(netdev);
 	netif_tx_disable(netdev);
 
+	if (adapter->capabilities & NX_FW_CAPABILITY_LINK_NOTIFICATION)
+		netxen_linkevent_request(adapter, 0);
+
 	if (adapter->stop_port)
 		adapter->stop_port(adapter);
 
-- 
1.6.3.3

